### ISSUE TITLE
Zone Occlusion: Ability to prioritize a zone

### ISSUE THREAD

**Issue by AleziaKurdis** [kasenvr/project-athena#351]
{Enhancement}

**Issue:**
Let say a typical case:
we have 2 zones, let’s name them: **OUTDOOR** and **INDOOR**
**INDOOR** is nested inside **OUTDOOR**
<img alt="image002" src="https://user-images.githubusercontent.com/60075796/82110090-bc6d4900-9709-11ea-89f6-e0fe562a1eb2.png">
Currently, If I stand in the "**INDOOR**" zone, for the zone occlusion I'm also considered as being in the "**OUTDOOR**" zone.
We do want this behavior in many cases, but in that specific example, this illustrates that it can be also a limitation.
If I'm in the "**INDOOR**" zone, I might not want to have everything from the "**OUTDOOR**" zone to be rendered, but only the entities that I flagged to be visible from the "**INDOOR**" zone.
To address that with the current solution, I’m forced to do complicated things like this:
<img alt="image004" src="https://user-images.githubusercontent.com/60075796/82110245-94cab080-970a-11ea-81b7-2e6be02f4161.png">
I will let you imagine how it would be in 3D,
And more evidently in a city where you would have many **"INDOOR"** nested in a **"OUTDOOR"**.
Quite difficult to manage and probably no the most performant way to proceed.

**A solution:**
Have a flag we can turn on on the Zone entity: "**Do not Inherit for Zone Occlusion**".
to force this behavior: "**Ignore other zones where I am standing**"
The rule would be: "**if this the flag, then ignore the zones over this current one**"

**Basically: "When I'm in that zone, don't count me in those that contain this one."**

By doing this, when the user will stand in the "**INDOOR**" zone (haveing that flag), the system won't consider him as standing in the "**OUTDOOR**" zone.  
Doing this would allow better and easier optimization of the rendering bandwidth.

**Comment by HifiExperiments** [comment]
ok I thought about this some more and maybe the easiest thing would be add a dontRenderWithZones property.  so in the case you drew, all the entities OUTSIDE would be marked renderWithZones OUTSIDE and dontRenderWithZones INSIDE, and then everything inside is just renderWithZones INSIDE.

does that make sense?  does that cover all the cases?

**Comment by AleziaKurdis** [comment]
This solution would work, but this solution has a significant disadvantage.
It's already something to mark all the entities that should be visible from a Zone.
But have to mark all the entities that are outside that must **not** be rendered could be a pain.
For each "INDOOR" zone, you would have to add an exclusion to a large part of the element of the "OUTDOOR" zone.

This is why I suggest instead to operate on the list of the zones where the avatar stand inside (the list used to figure when the entities need to be rendered depending if they are marked with at least one of these zones)

If we could flag a zone as "do not inherit",
When we get the list of the zones where the avatar stands inside
we would have to climb on that list from the last one and keep each of them until it hit one that is flagged.
You would get this way a subset that would exclude the larger zones over it.
And this subset would be used to determine if the entities should be rendered (according what is set in "renderWithZone")

![image](https://user-images.githubusercontent.com/60075796/82134615-68c13500-97c7-11ea-818e-abdbc2b362bb.png)

This way, no need to do massive exclusion manually.
Only one setup to make a zone behaving as the top one.
making literally a hole in a larger zone.

But well... I don't know how feasible this can be.

You can also compare this with what we have today for KeyLight
It's quite similar to "inherit" (where the Keyligh of the Larger zone is applied on the current)
and when we set "Yes" to the current, it uses it's own Keylight (which is a bit the equivalent... Making a hole in the larger zone to insert a different Keylight)
